Source/Modules/python.cxx

```diff
@@ -4536,6 +4540,18 @@ class PYTHON:public Language {
     Printf(f, "  };\n");
     Printv(f, "  PyObject *tuple_bases = SwigPyBuiltin_InitBases(bases);\n", NIL);
     Printf(f, "  PyTypeObject *pytype = (PyTypeObject *)PyType_FromSpecWithBases(&spec, tuple_bases);\n");
+    if (Getattr(n, "feature:python:bf_getbuffer")) {
+      Printv(f, "#if !defined(Py_LIMITED_API) && PY_VERSION_HEX < 0x03090000\n", NIL);
+      Printf(f, "  if (pytype) {\n");
+      Printf(f, "    *((PyTypeObject *)pytype)->tp_as_buffer = (PyBufferProcs){\n");
+      Printf(f, "      .bf_getbuffer = ");
+      Printv(f, getSlot(n, "feature:python:bf_getbuffer"), NIL);
+      Printf(f, ",\n      .bf_releasebuffer = ");
+      Printv(f, getSlot(n, "feature:python:bf_releasebuffer"), NIL);
+      Printf(f, "\n    };\n");
+      Printf(f, "  }\n");
```

Comment:
This results in code:
```c
#if !defined(Py_LIMITED_API) && PY_VERSION_HEX < 0x03090000
  if (pytype) {
    *((PyTypeObject *)pytype)->tp_as_buffer = (PyBufferProcs){
      .bf_getbuffer = ReadOnlyData::getbuffer,
      .bf_releasebuffer = ReadOnlyData::releasebuffer
    };
  }
#endif
```
1. Please switch to ISO c90 code, this is c99 syntax.
2. `(PyTypeObject *)` cast isn't necessary, pytype is already of type `PyTypeObject *`.
3. Isn't `tp_as_buffer` NULL by default, so this will be undefined behaviour??

Reply:
I've been experimenting and have found that `pytype->tp_as_buffer` is not `NULL`. It gets set in `PyType_FromMetaclass` which is called by `SwigPyBuiltin_InitBases`. See https://github.com/python/cpython/blob/28937d3a21cf8168c853ae43374a8287c21f71c9/Objects/typeobject.c#L5441
I'm now even more confused.